deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/resource/EnvDeploys.java

```diff
@@ -227,13 +227,24 @@ public Response create(
                     buildBean.getArtifact_url()));
         }
         // if the stage is not allowed (allow_private_build)
-        if(! envBean.getAllow_private_build()) { 
+        if(! envBean.getAllow_private_build()) {
             // only allow deploy if it is not private build
             if (buildBean.getScm_branch().equals("private")) {
                 throw new TeletaanInternalException(Response.Status.BAD_REQUEST, 
                     "This stage does not allow deploying a private build. Please Contact #teletraan to allow your stage for deploying private build");
             }
         }
+        // disallow sox deploy if the build artifact is private
+        if(envBean.getIs_sox() && buildBean.getScm_branch().equals("private")) {
```

Comment:
If by any change isSox can be null you should use Boolean.True.equals(envBean.getIs_sox())
that will prevent null issues.

Reply:
Thanks! It's not nullable with a non-null default, so should be safe.